v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java

```diff
@@ -189,6 +189,54 @@ public void migrationTestWithInsertsOnly() {
     assertMovieTableContents();
   }
 
+  @Test
+  public void interleavedAndFKAndIndexTest() {
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    uploadDataStreamFile(
+                        jobInfo,
+                        "Articles",
+                        "mysql-Articles.jsonl",
+                        "DataStreamToSpannerSimpleIT/mysql-Articles.jsonl"),
+                    uploadDataStreamFile(
+                        jobInfo,
+                        "Authors",
+                        "mysql-Authors.jsonl",
+                        "DataStreamToSpannerSimpleIT/mysql-Authors.jsonl"),
+                    uploadDataStreamFile(
+                        jobInfo,
+                        "Books",
+                        "mysql-Books.jsonl",
+                        "DataStreamToSpannerSimpleIT/mysql-Books.jsonl"),
+                    SpannerRowsCheck.builder(spannerResourceManager, "Articles")
+                        .setMinRows(3)
+                        .setMaxRows(3)
+                        .build(),
+                    SpannerRowsCheck.builder(spannerResourceManager, "Books")
+                        .setMinRows(3)
+                        .setMaxRows(3)
+                        .build(),
+                    SpannerRowsCheck.builder(spannerResourceManager, "Authors")
+                        .setMinRows(3)
+                        .setMaxRows(3)
+                        .build()))
+            .build();
+
```

Comment:
We should also add unhappy test case of trying to insert record in child table before inserting in parent table

Reply:
In that case shouldn't the insertion of child records fail?